onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -571,6 +571,7 @@ void RegisterExecutionProviders(InferenceSession* sess, const std::vector<std::s
       RegisterExecutionProvider(
           sess, *onnxruntime::CreateExecutionProviderFactory_ArmNN(sess->GetSessionOptions().enable_cpu_mem_arena));
 #endif
+    } else if (type == kDmlExecutionProvider) {
```

Comment:
Preventing Unknown provider type failure for the special cased ML op test. #4630 will fill in this placeholder.